zerver/actions/message_edit.py

```diff
@@ -1565,7 +1566,12 @@ def check_update_message(
     if isinstance(message_edit_request, StreamMessageEditRequest):
         if message_edit_request.is_stream_edited:
             assert message.is_stream_message()
-            if not user_profile.can_move_messages_between_streams():
+            if (
```

Comment:
I think here also the realm setting can be inside `can_move_messages_out_of_channel` function.